doc/rtd/topics/datasources/nocloud.rst

```diff
@@ -32,15 +32,84 @@ The permitted keys are:
 
 With ``ds=nocloud``, the ``seedfrom`` value must start with ``/`` or
 ``file://``.  With ``ds=nocloud-net``, the ``seedfrom`` value must start
-with ``http://`` or ``https://``.
+with ``http://`` or ``https://`` and end with a trailing ``/``.
+
+Cloud-init performs variable expansion of the seedfrom URL for any DMI kernel
+variables present in ``/sys/class/dmi/id`` or FreeBSD's kenv.
+Your ``seedfrom`` URL can contain variable names of the format `:dmi.varname:`
+to indicate to cloud-init NoCloud datasource that variable expansion to the
+actual value of the DMI system attribute is wanted. The leading and trailing
+colon `:` before and after any `:dmi.<varname>:` are used because other
```

Comment:
I think we can remove this sentence. It's a design detail that isn't really relevant to using it. If you decide to keep this sentence, `have` is misspelled on the following line and we have the same issue of docs say `:` but implementation uses `%`